### PR TITLE
[https://nvbugs/6185234][fix] register deepseek_v32 / kimi_k2 with transformers AutoConfig

### PR DESCRIPTION
**Pull request by longlee0622**
transformers 5.5.3 (PR #13994) no longer carries default `max_position_embeddings` on bare `PreTrainedConfig`. For model types that TRT-LLM understands locally but upstream transformers doesn't recognise (deepseek_v32, kimi_k2 — DeepSeek-V3.2 / Kimi-K2 checkpoints), `AutoConfig.from_pretrained` falls through to the bare config, and `AutoTokenizer.from_pretrained` then raises
`'PreTrainedConfig' object has no attribute 'max_position_embeddings'` before any tokenizer is constructed.

Caught by GB200 disagg perf-sanity
`disagg_upload-gen_only-gb200_deepseek-v32-fp4_1k1k_con1_ctx1_dep4_gen1_tep8_eplb0_mtp3_ccb-NIXL` in L0_PostMerge #2722: the `benchmark_serving` subprocess prints argparse `Namespace(...)` and exits 1 on the very next `get_tokenizer -> AutoTokenizer.from_pretrained(... trust_remote_code=True)` call. CTX/GEN servers log the same warning via `load_hf_tokenizer`. Pre-merge missed it because pre-merge DeepSeek-V3.2 coverage (TestDeepSeekV32 on DGX-B200) goes through `LLM(...)` which has tokenizer-load fallbacks, and only the post-merge GB200 perf-sanity stage runs `trtllm-serve` + `benchmark_serving`.

Fix: register `deepseek_v32` and `kimi_k2` against the local `DeepseekV3Config` in `transformers.models.auto.configuration_auto.CONFIG_MAPPING` at TRT-LLM import time, mirroring the existing `_CONFIG_REGISTRY` in `tensorrt_llm/_torch/pyexecutor/config_utils.py`. Registration goes through the underlying mapping (not `AutoConfig.register`) to bypass its `config.model_type == key` consistency check — `DeepseekV3Config` keeps `model_type = "deepseek_v3"` so existing in-engine paths are unaffected.

Trigger is placed in `tensorrt_llm/_torch/models/__init__.py` (already eagerly imported from `tensorrt_llm/__init__.py`) to avoid adding to `tensorrt_llm/__init__.py`'s E402 legacy-ruff baseline.

Pre-merge regression coverage added in
`tests/unittest/_torch/test_custom_config_registration.py`: both registration presence and an end-to-end
`AutoConfig.from_pretrained(<tmpdir config.json with model_type>)` call that hits the exact dispatch the broken post-merge test triggered.

## Summary by CodeRabbit

* **New Features**
  * Added support for `deepseek_v32` and `kimi_k2` model types with Transformers' AutoConfig and AutoTokenizer.
  * Improved compatibility with Transformers 5.5+.

* **Tests**
  * Added regression tests for model type registration.

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/TensorRT-LLM/pull/14293?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by Hudayday** [comment]
/bot run --disable-fail-fast --extra-stage "DGX_B200-8_GPUs-PyTorch-1, DGX_B200-8_GPUs-PyTorch-2, DGX_B200-8_GPUs-AutoDeploy-Post-Merge-1, DGX_B200-8_GPUs-PyTorch-PerfSanity-Post-Merge-4"

**Review by symphonylyh — APPROVED** [review]
should the filename change from `tensorrt_llm/_torch/configs/deepseek_v3.py` to something like `non_hf.py'? now it's not just DS v3 already, and later it may expand to more.
Approve for now as it's not a hard requirement. pls decide whether it makes sense

File: tensorrt_llm/_torch/configs/__init__.py
Comment (symphonylyh) [inline tensorrt_llm/_torch/configs/__init__.py:1]: should the filename change from tensorrt_llm/_torch/configs/deepseek_v3.py to something like `non_hf.py'? now it's not just DS v3 already, and later it may expand to more.
Approve for now as it's not a hard requirement. pls decide whether it makes sense

**Review by symphonylyh — APPROVED** [review]

File: tensorrt_llm/_torch/configs/__init__.py
Comment (longlee0622) [reply tensorrt_llm/_torch/configs/__init__.py:1]: thanks for the suggestion. I went ahead to merge PR. Will consider renaming file nezt time we touch it
